docs/examples/word_embedding/word_embedding_training.md

```diff
@@ -0,0 +1,374 @@
+                mx.nd.waitall()
+                wps = num_samples / (time.time() - start_time)
+                print('epoch %d, time %.2fs, iteration %d, throughput=%.2fK wps'
+                      % (epoch, time.time() - start_time, i, wps / 1000))
+
+        print('epoch %d, time %.2fs, train loss %.2f'
+              % (epoch, time.time() - start_time,
+                 train_l_sum.asscalar() / num_samples))
+        get_k_closest_tokens(vocab, embedding, 10, example_token)
+        print("")
+```
+
+```{.python .input}
+train_embedding(num_epochs=1)
+```
+
+### Word Similarity and Relatedness Task
+
+Word embeddings should capture the
+relationsship between words in natural language.
+In the Word Similarity and
+Relatedness Task word embeddings are evaluated by comparing word similarity
+scores computed from a pair of words with human labels for the similarity or
+relatedness of the pair.
+
+`gluonnlp` includes a number of common datasets for
+the Word Similarity and Relatedness Task. The included datasets are listed in
+the [API documentation](http://gluon-nlp.mxnet.io/api/data.html#word-embedding-
+evaluation-datasets). We use several of them in the evaluation example below.
+We first show a few samples from the WordSim353 dataset, to get an overall
+feeling of the Dataset structur
+
+### Evaluation
+
+Thanks to the subword support of the `FasttextEmbeddingModel` we
+can evaluate on all words in the evaluation dataset,
+not only the ones that we
+observed during training.
+
+We first compute a list of tokens in our evaluation
+dataset and then create a embedding matrix for them based on the fastText model.
+
+```{.python .input}
+rw = nlp.data.RareWords()
+rw_tokens  = list(set(itertools.chain.from_iterable((d[0], d[1]) for d in rw)))
+
+rw_token_embedding = nlp.embedding.TokenEmbedding(unknown_token=None, allow_extend=True)
+rw_token_embedding[rw_tokens]= embedding[rw_tokens]
+
+print('There are', len(rw_tokens), 'unique tokens in the RareWords dataset. Examples are:')
+for i in range(5):
+    print('\t', rw[i])
+print('The imputed TokenEmbedding has shape', rw_token_embedding.idx_to_vec.shape)
+```
+
+```{.python .input}
+evaluator = nlp.embedding.evaluation.WordEmbeddingSimilarity(
+    idx_to_vec=rw_token_embedding.idx_to_vec,
+    similarity_function="CosineSimilarity")
+evaluator.initialize(ctx=context)
+evaluator.hybridize()
+```
+
+```{.python .input}
+words1, words2, scores = zip(*([rw_token_embedding.token_to_idx[d[0]],
+                                rw_token_embedding.token_to_idx[d[1]],
+                                d[2]] for d in rw))
+words1 = mx.nd.array(words1, ctx=context)
+words2 = mx.nd.array(words2, ctx=context)
+```
+
+```{.python .input}
+pred_similarity = evaluator(words1, words2)
+sr = stats.spearmanr(pred_similarity.asnumpy(), np.array(scores))
+print('Spearman rank correlation on {} pairs of {}: {}'.format(
+    len(words1), rw.__class__.__name__, sr.correlation.round(3)))
+```
+
+# Further information
```

Comment:
Title level is incorrect, as it's showing up on the main page. http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-384/20/examples/word_embedding/word_embedding_training.html#Further-information